src/uct/base/uct_iface.c

```diff
@@ -555,9 +555,30 @@ ucs_status_t uct_cm_client_ep_conn_notify(uct_ep_h ep)
     return ep->iface->ops.cm_ep_conn_notify(ep);
 }
 
+ucs_status_t uct_base_ep_stats_reset(uct_base_ep_t *ep, uct_base_iface_t *iface)
+{
+    ucs_status_t status;
+
+    UCS_STATS_NODE_FREE(ep->stats);
+
+    status = UCS_STATS_NODE_ALLOC(&ep->stats, &uct_ep_stats_class, iface->stats,
+                                  "-%p", ep);
+#ifdef ENABLE_STATS
```

Comment:
why need ifdef?

Reply:
I guess for `ep->stats = NULL;`

Reply:
right, 'release' mode doesn't compile otherwise.

Reply:
how does it work in other places where we use UCS_STATS_NODE_ALLOC?

Reply:
and what is the reason to set ep->stats=NULL if allocation fails?

Reply:
so that if this allocation fails when this reset function is called, but the free in it succeeds, another free call won't fail when the cleanup flow is called on the error flow